ufl/domain.py

```diff
@@ -50,6 +58,33 @@ def topological_dimension(self):
         """Return the dimension of the topology of this domain."""
         return self._topological_dimension
 
+    @property
+    def meshes(self):
+        """Return the component meshes."""
+        raise NotImplementedError("meshes() method not implemented")
+
+    def __len__(self):
+        """Return number of component meshes."""
+        return len(self.meshes)
+
+    def __getitem__(self, i):
+        """Return i-th component mesh."""
+        if i >= len(self):
+            raise ValueError(f"index ({i}) >= num. component meshes ({len(self)})")
+        return self.meshes[i]
+
+    def __iter__(self):
+        """Return iterable component meshes."""
+        return iter(self.meshes)
+
+    def iterable_like(self, element: AbstractFiniteElement) -> Iterable["Mesh"] | "MeshSequence":
```

Comment:
Do we need to increase the minimum python version in pyproject.toml to  allow using `|` in typing?
Given the minimum version is currently 3.8 which has passed end-of-life, we should probably bump it anyway, but not necessarily in this PR unless it's needed here

Reply:
Thanks. It's probably not a good idea to bump python version in this PR. I just switched to `typing.Union`.